jwst/resample/resample_spec_step.py

```diff
@@ -108,23 +112,30 @@ def _process_slit(self, input_models):
 
         Returns
         -------
-        result : `~jwst.datamodels.DrizProductModel`
+        result : `~jwst.datamodels.ImageModel`
             The resampled output, one per source
         """
+
         resamp = resample_spec.ResampleSpecData(input_models, **self.drizpars)
-        drizzled_models = resamp.do_drizzle()
-
-        for model in drizzled_models:
-            model.meta.cal_step.resample = "COMPLETE"
-            model.meta.asn.pool_name = input_models.meta.pool_name
-            model.meta.asn.table_name = input_models.meta.table_name
-            update_s_region_spectral(model)
-
-        # Return either the single resampled datamodel, or the container
-        # of datamodels.
-        if len(drizzled_models) == 1:
-            result = drizzled_models[0]
+
+        if input_models[0].meta.exposure.type == "MIR_LRS-FIXEDSLIT":
+            bb = input_models[0].meta.wcs.bounding_box
+            ((x1, x2), (y1, y2)) = bb
+            xmin = int(min(x1, x2))
+            ymin = int(min(y1, y2))
+            xmax = int(max(x1, x2))
+            ymax = int(max(y1, y2))
+            drizzled_models = resamp.do_drizzle(xmin=xmin, xmax=xmax,
+                                                ymin=ymin, ymax=ymax)
         else:
-            result = drizzled_models
+            drizzled_models = resamp.do_drizzle()
 
+        drizzled_models[0].update(input_models[0], only="PRIMARY")
```

Comment:
These 2 lines doing updates are causing failures in the regtests for some of the exposure time related keywords in the output (resampled) image header. The drizzle routines compute new values for some of them, such as the total exposure time of all images that've been combined and put those values into the `drizzled_models` that's returned. But then these 2 update calls essentially reset all of the meta data back to their original input values, which is causing the differences in the regtests (e.g. EFFEXPTM is back to the original single exposure values, rather than the total). The original `process_slit` code did not do any update like this on the returned `drizzled_models`, so we shouldn't need it in this new version either. When I remove these lines (133-134), s2d products from `calwebb_spec3` go back to not showing any differences in the regtest comparisons.